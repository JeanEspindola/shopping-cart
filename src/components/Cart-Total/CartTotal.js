/*
* Cart Total Presentation Component.
*/
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Row, Col, Glyphicon } from 'react-bootstrap';
import cartListService from '../../utils/cartListService';
import './CartTotal.css';
import initialState from '../../reducers/initialState';

let totalResults = initialState.total;

class CartTotal extends Component {
  componentWillReceiveProps(newProps) {
    if (newProps.list.length !== 0 && newProps.list !== this.props.list) {
      totalResults = cartListService.getTotalValues(newProps.list);
    }
  }

  render() {
    return (
      <div className="total">
        <Row className="total__net">
          <Col xs={6}>Net Total</Col>
          <Col xs={6} className="total__values">{totalResults.netTotal}€</Col>
        </Row>
        <hr className="total__divider" />
        <Row className="total__tax">
          <Col xs={6}>Tax</Col>
          <Col xs={6} className="total__values">{totalResults.taxTotal}€</Col>
        </Row>
        <Row className="total__tax__7">
          <Col xs={6}><Glyphicon glyph="arrow-right" /> 7%</Col>
          <Col xs={6} className="total__values">{totalResults.tax7}€</Col>
        </Row>
        <Row className="total__tax__19">
          <Col xs={6}><Glyphicon glyph="arrow-right" />19%</Col>
          <Col xs={6} className="total__values">{totalResults.tax19}€</Col>
        </Row>
        <hr className="total__divider" />
        <Row className="total__grand">
          <Col xs={6}>Grand Total</Col>
          <Col xs={6} className="total__values">{totalResults.grandTotal}€</Col>
        </Row>
      </div>
    );
  }
}

CartTotal.propTypes = {
  list: PropTypes.array.isRequired,
};

export default CartTotal;
