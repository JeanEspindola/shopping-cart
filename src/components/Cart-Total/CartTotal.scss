.total {
  background-color: white;
  float: right;
  width: 250px;
}

.empty {
  height: 35px;
}

.total__divider {
  margin-bottom: 10px;
  margin-top: 10px;
}

.total__net {
  margin: 0 5px 0 5px;
  padding-top: 10px;
}

.total__grand {
  margin: 0 5px 0 5px;
  color: blue;
  padding-bottom: 10px;
}

.total__tax {
  margin: 0 5px 5px 5px;
}

.total__tax__7, .total__tax__19 {
  color: gray;
  margin: 0 5px 0 5px;
}

.total__values {
  text-align: right;
}
