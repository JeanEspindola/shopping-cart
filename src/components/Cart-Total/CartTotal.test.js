import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import CartTotal from './CartTotal';

describe('CartTotal', () => {
  let list;
  let wrapper;

  it('renders CartTotal without crashing', () => {
    list = [];
    wrapper = shallow(<CartTotal list={list} />);
    expect(wrapper.length).toEqual(1);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
