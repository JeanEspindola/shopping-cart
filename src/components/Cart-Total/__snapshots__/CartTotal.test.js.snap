// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartTotal renders CartTotal without crashing 1`] = `
<div
  className="total empty"
>
  <Row
    bsClass="row"
    className="total__net"
    componentClass="div"
  >
    <Col
      bsClass="col"
      componentClass="div"
      id="total"
      xs={6}
    >
      Net Total
    </Col>
    <Col
      bsClass="col"
      className="total__values"
      componentClass="div"
      id="value"
      xs={6}
    >
      0â‚¬
    </Col>
  </Row>
</div>
`;
